### ISSUE TITLE
Opening jupiter lab/notebook doesn't auto-launch into browser but gets stuck on unhelpful "opening jupyter application page"

### ISSUE THREAD

**Issue by drmikecooke** [jupyterlab/jupyterlab#16221]
I am using a raspberry pi4 with 64bit bookworm os (installed in the past few days).

I have used jupyter lab before, but pip installed new version (4.1.6) in a virtual environment (venv) and I can get it to work after quitting the unhelpful "opening jupyter application page" (curses-type menu interface using keyboard arrows etc.) and doing a right click "open URL" on the tokenized URLs showing in terminal.

The unhelpful page also pops up in notebook launch.

I thought it might be a firewall problem (ufw) but the rotten page popped up when ufw was disabled.

The unhelpful page seems to want me to set up cookies and so on, but I don't really understand why they are involved.

Not sure what else to add.

**Comment by drmikecooke** [comment]
Hi there https://github.com/JasonWeill,

![20240424_08h53m43s_grim](https://github.com/jupyterlab/jupyterlab/assets/65656132/b4e38ebf-0197-43f8-bdb2-87de03b9f95c)

Thanks for getting back to me so quickly. Hopefully the image above will help in understanding my problem. What you see is a request about cookies and then a message from the server about the skipped servers, and that the build is up to date.

Before (bullseye) I used to get these messages continuously, and once I started work in the auto-launched browser window, there would be a log of my actions in jlab.

Answering the cookie question "A" for always, allows me to continue suggesting I may want "help", which gets me:

![20240424_09h03m41s_grim](https://github.com/jupyterlab/jupyterlab/assets/65656132/8822fba4-e80f-41fe-b1ac-94f47e0f0a6a)

If you know the code it may tell you what's happening. It doesn't tell me much of use.

If I now quit "q" I get back to the traditional log.

As a bit of extra information I have set up a bookmark in my browser, which seems to work if I just launch jupyter-lab without me having to right click in the terminal.

Hope there are enough breadcrumbs here.

Mike

**Comment by JasonWeill** [comment]
@drmikecooke Aha! In your first screen shot, that cookie warning against a blue background is coming from a text mode browser, like Lynx or Links, that seems to be detected as your system's default web browser. JupyterLab won't work in a text mode browser; you need to configure it to spawn a GUI browser like Chromium.

You can set the `BROWSER` environment variable, or the `c.ServerApp.browser` configuration setting ([docs](https://jupyter-server.readthedocs.io/en/stable/api/jupyter_server.html#jupyter_server.serverapp.ServerApp.browser)) to force a particular app to launch as your default web browser. Please let me know if that works for you!

**Comment by drmikecooke** [comment]
I tried both, and both worked.

Had to generate a configuration to access c.ServerApp.browser:

```bash
jupyter server --generate-config
```

and editing the file ~/.jupyter/jupyter_server_config.py to uncomment the line (226 in the version I have):

c.ServerApp.browser = 'chromium-browser'

The environment variable is set using:

```bash
BROWSER=chromium-browser
export BROWSER
```

For this option to be permanent, I guess you would have to put these lines into .bashrc (for command-line launch of jupyter-lab) and .profile (for launch from menu without terminal).

Of course, if users want to use firefox or something else, that is up to them.

I'm not sure why the system (bookworm) is defaulting to lynx rather than chromium. It's a relatively new debian release, so I guess it is a little rough around the edges?

Thanx anyway Jason for helping me out here. I guess the thread is now closed? Is that something I should do?

Mike
